cmake/PrivateSdlFunctions.cmake

```diff
@@ -218,3 +218,26 @@ macro(sdl_check_project_in_subfolder relative_subfolder name)
             "Run the download script in the external folder, or re-configure with -DSDL2IMAGE_VENDORED=OFF to use system packages.")
     endif()
 endmacro()
+
+macro(sdl_check_linker_flag flag var)
+    # FIXME: Use CheckLinkerFlag module once cmake minimum version >= 3.18
+    include(CMakePushCheckState)
+    include(CheckCSourceCompiles)
+    cmake_push_check_state(RESET)
+    set(CMAKE_REQUIRED_LINK_OPTIONS "-Wl,--no-undefined")
```

Comment:
I'm OK with this to get the functionality in. But: Can't this helper macro get an additional arg instead of hard-coding `-Wl,--no-undefined` here again? (sorry, not good at cmake.)

Reply:
I'm afraid not.
The `check_c_source_compiles` is itself a wrapper around [`try_compile`](https://cmake.org/cmake/help/latest/command/try_compile.html#command:try_compile).
See https://cmake.org/cmake/help/latest/module/CheckCSourceCompiles.html

`try_compile` accepts a `LINK_OPTIONS` argument, but it also requires additional arguments that would complicate the use here, such as bindir, and the need of saving the test source to disk ourselves.
Using the convenience function `check_c_source_compiles` also caches the result.

Reply:
OK then, taking this as is.